sklearn/metrics/_classification.py

```diff
@@ -1481,29 +1548,31 @@ def precision_recall_fscore_support(
     sample_weight : array-like of shape (n_samples,), default=None
         Sample weights.
 
-    zero_division : "warn", 0 or 1, default="warn"
+    zero_division : "warn", 0, 1 or np.nan, default="warn"
         Sets the value to return when there is a zero division:
            - recall: when there are no positive labels
            - precision: when there are no positive predictions
            - f-score: both
 
-        If set to "warn", this acts as 0, but warnings are also raised.
+        Notes:
+        - If set to "warn", this acts like 0, but a warning is also raised.
+        - If set to np.nan, such values will be excluded from the average.
 
     Returns
     -------
-    precision : float (if average is not None) or array of float, shape =\
+    precision : float (if `average` is not None) or array of float, shape =\
```

Comment:
There is already a lot moving parts in this PR. Can this change be a separate small PR?